src/Containers/Stock.tsx

```diff
@@ -0,0 +1,122 @@
+import { ArrowUp } from '@styled-icons/entypo/ArrowUp';
+import { ArrowDown } from '@styled-icons/entypo/ArrowDown';
+import { flex } from '../Utils/Mixins';
+import {
+    Main,
+    Responsive,
+    MainInterface,
+    ResponsiveInterface,
+} from '../Utils/BaseStyles';
+import { MainTheme } from '../Themes';
+
+interface ChartDataItem {
+    value: string | number;
+}
+
+export interface StockProps extends StockBoxProps {
+    chartData: Array<ChartDataItem>;
+    chartColor: string;
+    title: string;
+    figure: number;
+    rate: number;
+    bgColor: string;
+}
+
+const getChartColor = (colorKey: string) =>
+    MainTheme.colors[colorKey] || colorKey;
+
+const getFigure = (figure: number) =>
+    figure.toLocaleString(undefined, { maximumFractionDigits: 2 });
+
+const getRateIcon = (rate: number) =>
+    rate < 0 ? <StyledArrowDown /> : <StyledArrowUp />;
+
+export const Stock: React.FC<StockProps> = ({
+    chartColor,
+    chartData,
+    title,
+    figure,
+    rate,
+    ...props
+}): React.ReactElement => {
+    return (
+        <StockBox {...props}>
+            <Content>
+                <Text size="h6" color="text">
+                    <strong>{title}</strong>
+                </Text>
+                <Text size="h1" color="background">
+                    <strong>{getFigure(figure)}</strong>
+                </Text>
+                <Text size="h6" color={chartColor}>
+                    <strong>
+                        {`${getFigure(rate)} `}
+                        {getRateIcon(rate)}
+                    </strong>
+                </Text>
+            </Content>
+            <Content>
+                <LineChart width={150} height={50} data={chartData}>
+                    <Line
+                        type="linear"
+                        dataKey="value"
+                        stroke={getChartColor(chartColor)}
+                        strokeWidth={2}
+                        dot={false}
+                    />
+                </LineChart>
+            </Content>
+        </StockBox>
+    );
+};
+
+interface StockBoxProps
+    extends MainInterface,
+        ResponsiveInterface,
+        React.HTMLAttributes<HTMLDivElement> {
+    bgColor: string;
+}
+
+const StockBox = styled.div<StockBoxProps>`
+    // Theme Stuff
+    ${({ theme, ...props }): string => `
+    border-radius: ${theme.dimensions.radius};
+    font-family: ${theme.font.family};
+    color: ${theme.colors.text};
+    background-color:${MainTheme.colors[props.bgColor] || props.bgColor};
+    ${Main({
+        ...props,
+    })}
+    `}
+    // Base Styles
+    ${Responsive}
+    ${flex('flex-start', 'center')}
+    width:fit-content;
+`;
+
+interface TextProps extends React.HTMLAttributes<HTMLDivElement> {
+    size: string;
+}
+
+const Text = styled.div<TextProps>`
+    ${({ theme, ...props }): string => `
+font-size:${theme.font.size[props.size] || props.size};
+color:${theme.colors[props.color as string] || props.color};
+`}
+`;
```

Comment:
There are three different Text elements in the React-UI library: Paragraph, SmallText, and Heading. No need to create a text element for each component you make. Use other components from the library when you can.

Reply:
I know. But these you mention are all p tags in the end, and a p tag has a lot of margins vertically, and I needed less margin, so I implemented my own version for this reason. I remember you already commented on this point to me in the previous PR.

Reply:
With the type attribute the components can be typed as div.
![Type change](https://user-images.githubusercontent.com/29719870/95141908-3b9a0280-0727-11eb-8620-51a6dc984dd5.png)
![Screenshot from 2020-10-05 16-24-32](https://user-images.githubusercontent.com/29719870/95141926-45236a80-0727-11eb-8add-11217d192d22.png)